src/fsharp/NameResolution.fs

```diff
@@ -2253,7 +2253,14 @@ let rec ResolveLongIdentInTypePrim (ncenv: NameResolver) nenv lookupKind (resInf
                         addToBuffer uc.DisplayName
             | _ -> ()
 
-        raze (UndefinedName (depth, FSComp.SR.undefinedNameFieldConstructorOrMember, id, suggestMembers))
+        let errorTextF s =
+            if isAppTy g ty then
```

Comment:
that's a nice trick! we should do it for all FS039

Reply:
Yeah, good point. I'll take a look ... once I can figure out the right fsharpqa tests to adjust